test/unit/test_extension_wrapper_common.py

```diff
@@ -0,0 +1,81 @@
+import unittest.mock
+import pytest
+import tempfile
+
+from exasol.nb_connector.secret_store import Secrets
+from exasol.nb_connector.ai_lab_config import AILabConfig as CKey
+from exasol.nb_connector.extension_wrapper_common import encapsulate_bucketfs_credentials
+
+
+@pytest.fixture
+def filled_secrets(secrets) -> Secrets:
+    secrets.save(CKey.db_host_name, 'localhost')
+    secrets.save(CKey.db_port, '8888')
+    secrets.save(CKey.db_user, 'user')
+    secrets.save(CKey.db_password, 'password')
+    secrets.save(CKey.bfs_port, '6666')
+    secrets.save(CKey.bfs_encryption, 'True')
+    secrets.save(CKey.bfs_service, 'bfsdefault')
+    secrets.save(CKey.bfs_bucket, 'default')
+    secrets.save(CKey.bfs_user, 'user'),
+    secrets.save(CKey.bfs_password, 'password')
+    return secrets
+
+
+@unittest.mock.patch("pyexasol.connect")
+def test_bucketfs_credentials_default(mock_connect, filled_secrets):
+
+    path_in_bucket = 'location'
+
+    mock_connection = unittest.mock.MagicMock()
+    mock_connection.__enter__.return_value = mock_connection
+    mock_connect.return_value = mock_connection
+
+    encapsulate_bucketfs_credentials(filled_secrets, path_in_bucket=path_in_bucket,
+                                     connection_name='whatever')
+    expected_url = f"https://localhost:6666/default/{path_in_bucket};bfsdefault"
+
+    mock_connection.execute.assert_called_once()
+    query = mock_connection.execute.call_args_list[0].kwargs['query']
+    assert f"TO '{expected_url}'" in query
+
+
+@unittest.mock.patch("pyexasol.connect")
+def test_bucketfs_credentials_not_verify(mock_connect, filled_secrets):
+
+    path_in_bucket = 'location'
+    filled_secrets.save(CKey.cert_vld, 'no')
+
+    mock_connection = unittest.mock.MagicMock()
+    mock_connection.__enter__.return_value = mock_connection
+    mock_connect.return_value = mock_connection
+
+    encapsulate_bucketfs_credentials(filled_secrets, path_in_bucket=path_in_bucket,
+                                     connection_name='whatever')
+    expected_url = f"https://localhost:6666/default/{path_in_bucket};bfsdefault#False"
+
+    mock_connection.execute.assert_called_once()
+    query = mock_connection.execute.call_args_list[0].kwargs['query']
+    assert f"TO '{expected_url}'" in query
+
+
+@unittest.mock.patch("pyexasol.connect")
+def test_bucketfs_credentials_ca(mock_connect, filled_secrets):
+
+    with tempfile.NamedTemporaryFile() as f:
```

Comment:
You are only creating a CREATE CONNECTION, so, I think, you don't need to create a real file, or does encapsulate_bucketfs_credentials checks if the file exists. However, this reminds me that the file might be not at the same place in the UDFs, or it might not be available there. So, we might be need to handle this case. Either, we upload the certificate into the bucketfs from the AI-Lab and use this path in the connection or the user need to specify the path.

Reply:
The way it's implemented, which is a bit unfortunate, is that it does check that the file exists when it opens the pyexasol connection. The purpose is dual. Firstly it validates the path. Secondly, it checks if it's a file or a directory. Depending on that different parameters are set in the SSL structure.

I see your point. It's a bit tricky situation. We need potentially two different sets of communication settings.
1. To run the `CREATE CONNECTION ...`  command from the local machine or wherever we are running the notebook connector.
2. Executing commands from UDF or wherever this connection object is consumed.
Unfortunately, we have only one set of parameters in the AI-Lab config.

Maybe the logic should go like this:
If the user wants to have the verification on (verify = True) - fine, presumably the server has a valid certificate, recognized with the default set of CAs.
If the user provides a CA file or directory - oops, we can't do this from the UDF, let's turn the verification off instead. This is far from ideal, but I it should make the situation better than it is now.

Reply:
I like your suggestion, it is a good compromise for the moment. My idea, with uploading the certificate to the bucketfs, could stay as a feature ticket, but it also contains some security concerns. The DB and UDF actually need a proper trust store handling.